flang/lib/Lower/Bridge.cpp

```diff
@@ -1473,7 +1474,25 @@ class FirConverter : public Fortran::lower::AbstractConverter {
   /// the global to the ModuleOp as a new uniqued symbol and initialize it with
   /// the correct value. It will be referenced on demand using `fir.addr_of`.
   void instantiateGlobal(const Fortran::lower::pft::Variable &var) {
-    llvm_unreachable("Varun: put your code here");
+    const auto &sym = var.getSymbol();
+    std::string globalName = uniqueCGIdent(sym.name().ToString());
+    auto global = builder->getNamedGlobal(globalName);
+    if (!global) {
+      if (const auto *details =
+              sym.detailsIf<Fortran::semantics::ObjectEntityDetails>()) {
+        if (details->init()) {
+          global = builder->createGlobal(
+              toLocation(), genType(sym), globalName, false,
+              [&](Fortran::lower::FirOpBuilder &builder) {
+                auto initVal = genExprValue(details->init().value());
```

Comment:
OK, that is a tricky question (maybe aimed at Eric ;) ), is this `genExprValue` using the right builder ? Expression conversion will use the builder from the `AbstractConverter`, but is it the same as the builder provided here (used inside the global region) ?

Reply:
It should be using the same builder. (`genExprValue` just forwards the "current location" as being tracked by the bridge, but otherwise is a pass-through.)